x-pack/plugins/security_solution/public/flyout/right/hooks/use_correlations.ts

```diff
@@ -125,47 +148,72 @@ export const useCorrelations = ({
     loading: sameSourceAlertsLoading,
     error: sameSourceAlertsError,
     dataCount: sameSourceAlertsCount,
+    data: sameSourceAlertsIds,
   } = useFetchRelatedAlertsBySameSourceEvent({
     dataFormattedForFieldBrowser,
     scopeId,
   });
-  if (showSameSourceAlerts && !sameSourceAlertsLoading && !sameSourceAlertsError) {
-    data.push({
-      icon: 'warning',
-      value: sameSourceAlertsCount,
-      text:
-        sameSourceAlertsCount <= 1
-          ? CORRELATIONS_SAME_SOURCE_EVENT_ALERT
-          : CORRELATIONS_SAME_SOURCE_EVENT_ALERTS,
-    });
-  }
+
+  useEffect(() => {
+    if (showSameSourceAlerts && !sameSourceAlertsLoading && !sameSourceAlertsError) {
+      updateInsightsSummaryPanel({
+        icon: 'warning',
+        value: sameSourceAlertsCount,
+        text:
+          sameSourceAlertsCount <= 1
+            ? CORRELATIONS_SAME_SOURCE_EVENT_ALERT
+            : CORRELATIONS_SAME_SOURCE_EVENT_ALERTS,
+      });
+    }
+  }, [sameSourceAlertsCount, sameSourceAlertsError, sameSourceAlertsLoading, showSameSourceAlerts]);
 
   // alerts related by session
   const showAlertsBySession = useShowRelatedAlertsBySession({ dataFormattedForFieldBrowser });
   const {
     loading: alertsBySessionLoading,
     error: alertsBySessionError,
     dataCount: alertsBySessionCount,
+    data: alertsBySessionIds,
   } = useFetchRelatedAlertsBySession({
     dataFormattedForFieldBrowser,
     scopeId,
   });
-  if (showAlertsBySession && !alertsBySessionLoading && !alertsBySessionError) {
-    data.push({
-      icon: 'warning',
-      value: alertsBySessionCount,
-      text:
-        alertsBySessionCount <= 1
-          ? CORRELATIONS_SAME_SESSION_ALERT
-          : CORRELATIONS_SAME_SESSION_ALERTS,
-    });
-  }
 
-  return {
-    loading:
-      casesLoading || ancestryAlertsLoading || alertsBySessionLoading || sameSourceAlertsLoading,
-    error: data.length === 0,
-    data: data || [],
-    dataCount: data.length || 0,
-  };
+  useEffect(() => {
+    if (showAlertsBySession && !alertsBySessionLoading && !alertsBySessionError) {
+      updateInsightsSummaryPanel({
+        icon: 'warning',
+        value: alertsBySessionCount,
+        text:
+          alertsBySessionCount <= 1
+            ? CORRELATIONS_SAME_SESSION_ALERT
+            : CORRELATIONS_SAME_SESSION_ALERTS,
+      });
+    }
+  }, [alertsBySessionCount, alertsBySessionError, alertsBySessionLoading, showAlertsBySession]);
+
+  return useMemo(
+    () => ({
+      loading:
+        casesLoading || ancestryAlertsLoading || alertsBySessionLoading || sameSourceAlertsLoading,
+      error: data.length === 0,
+      data: data || [],
+      dataCount: data.length || 0,
+      alertsBySessionIds,
+      sameSourceAlertsIds,
+      ancestryAlertsIds: ancestryAlertsIds || [],
+      cases: cases || [],
```

Comment:
what do you think for these 2 to move the `|| []` to the hooks that are used under the hood (respectively the x-pack/plugins/security_solution/public/flyout/right/hooks/use_fetch_related_alerts_by_ancestry.ts and [useFetchRelatedCases](https://github.com/elastic/kibana/blob/main/x-pack/plugins/security_solution/public/flyout/right/hooks/use_fetch_related_cases.ts#L62)) and modify the respective types from in the `Result` interfaces.

That would make everything more consistent. I guess that was on me when I created those hooks, sorry!